packages/esm-patient-chart-app/src/patient-flags/hooks/usePatientFlags.tsx

```diff
@@ -0,0 +1,90 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+interface FlagFetchResponse {
+  uuid: string;
+  message: string;
+  voided: boolean;
+  flag: { uuid: string; display: string };
+  patient: { uuid: string; display: string };
+  tags: Array<{ uuid: string; display: string }>;
+  auditInfo: { dateCreated: string };
+}
+
+interface FlagsFetchResponse {
+  results: Array<FlagFetchResponse>;
+}
+
+/**
+ * React hook that takes in a patient uuid and returns
+ * patient flags for that patient together with helper objects
+ * @param patientUuid Unique patient idenfier
+ * @returns An array of patient identifiers
+ */
+export function useFlagsFromPatient(patientUuid: string) {
+  const { data, error, isValidating, mutate } = useSWR<{ data: FlagsFetchResponse }, Error>(
+    `/ws/rest/v1/patientflags/patientflag?patient=${patientUuid}&v=full`,
+    openmrsFetch,
+  );
+  const result = useMemo(() => {
+    return {
+      flags: data?.data?.results ?? [],
+      flagLoadingError: error,
+      isLoadingFlags: !data && !error,
+      isValidatingFlags: isValidating,
+      mutate,
+    };
+  }, [data, error, isValidating, mutate]);
```

Comment:
Please don't use useMemo with isValidating, because the isValidating variable changes everytime user focuses out and in the window, if the data is stale and all. 
`useMemo` will run a lot of the caching logic for it to be re-cached again very frequently.